docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
```

Comment:
In other words, implement the [database/sql](https://golang.org/pkg/database/sql/) registration pattern for deps. Example [#1](https://github.com/coreos/clair/tree/master/worker/detectors) [#2](https://github.com/chihaya/chihaya/blob/master/server/server.go)